### PR TITLE
authentication service: basic openapi maven plugin integration

### PR DESCRIPTION
**Pull request by mrcmis**
as mentioned in team meeting, we can try to implement openapi/swagger to share schema and generate some code for us - this is just initial implementation. Classes will be generated after executing mvn clean install

### CODE REVIEW

**Comment by mrcmis** [comment]
@DorianKossowski

File: authentication-service/src/main/resources/api.yaml [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+x-samples-languages:
+  - java
+info:
+  description: Authentication service for WFIiS AGH game shop project
+  version: 1.0.0
+  title: Authentication Service
+tags:
+  - authentication
+paths:
+  /authenticate:
```
Comment (PiotrKolecki) [inline authentication-service/src/main/resources/api.yaml:11]: it this correct path?

**Review by DorianKossowski — APPROVED** [review]

File: authentication-service/src/main/resources/api.yaml (hunk 1 above)
Comment (mrcmis) [reply authentication-service/src/main/resources/api.yaml:11]: i think so, this is our endpoint path

**Review by PiotrKolecki — APPROVED** [review]
approved

**Pull request merged**